Feature/individual intent component

sarah-arrrgh opened the pull request:
closes #49

sarah-arrrgh committed:
feat(ordering): create storybook product list components

sarah-arrrgh committed:
feat(ordering): storybook list view product component

sarah-arrrgh committed:
feat(ordering): wire up single view product in storybook

sarah-arrrgh committed:
feat(ordering): scaffold product facet storybook component

TODO: style product facet component

sarah-arrrgh committed:
feat(ordering): wire up product price spec sub component in storybook

sarah-arrrgh committed:
feat(ordering): render basic price spec component in storybook

sarah-arrrgh committed:
feat(ordering): render price spec sub-component in single product view component in storybook

sarah-arrrgh committed:
feat(ordering): add add button button

ahdinosaur commented on ordering/components/ProductList.js:
```diff
@@ -0,0 +1,26 @@
+  return (
+    h('div', {
+      className: styles.container
+    },
```
i think even though `map` returns an array, this should all be contained within an `[]` (see [`react-hyperscript` source](https://github.com/mlmorg/react-hyperscript/blob/master/index.js) to understand what is happening). then returned react elements from `map` need `key: product.id`.

sarah-arrrgh replied:
done!

ahdinosaur commented on app/locales/en.json:
```diff
@@ -57,9 +57,10 @@
+  "ordering.from": "from",
+  "ordering.add": "add",
   "ordering.intentTitle": "{name} per {step, number} {unit}",
+  "ordering.priceSpec": "if everyone orders more than {minimum}, then this item will cost ${price} each, in which case, I want to order ",
```
sweet!

ahdinosaur commented on ordering/components/ListViewProduct.js:
```diff
@@ -0,0 +1,54 @@
+  )
+}
+
+export default pipe(
```
can we use `compose` instead of `pipe`? i'd like us to be consistent, they have opposite orderings. i started using `pipe`, but have since realized that `compose` makes more sense (also is provided by `recompose`).

sarah-arrrgh replied:
done!

ahdinosaur commented on ordering/components/SingleViewProduct.js:
```diff
@@ -0,0 +1,74 @@
+              id: 'ordering.from',
+              className: styles.fromText
+            }),
+            ` $${product.priceSpecifications[0].price}`
```
i reckon this should be part of the above internationalization text, so the message is like `ordering.price` and is passed in the price, like `from ${price}`, or something along those lines.

sarah-arrrgh replied:
have changed it so it will show as `NZD7.99` instead of `$7.99`

ahdinosaur commented on ordering/components/SingleViewProduct.js:
```diff
@@ -0,0 +1,74 @@
+        ]),
+        h('div', {
+          className: styles.priceSpecsContainer
+        }, map((priceSpec) => {
```
see previous comment about `map` and hyperscript.

ahdinosaur replied:
also, the more "ramda-y" way to do this would be:

```js
const renderPriceSpecs = map(priceSpec => (
  h(ProductPriceSpec, {
    priceSpec,
    key: priceSpec.id
  })
))

    // ...
    }, [
      renderPriceSpecs(product.priceSpecs)
    ])
    // ...
```

ahdinosaur commented on ordering/stories/ProductList.js:
```diff
@@ -0,0 +1,68 @@
+    name: 'crayons',
+    description: 'these are crayons',
+    image: 'http://www.mercurius-australia.com/site/images/1250623.jpg',
+    priceSpecifications: [
```
i changed this to `priceSpecs` (because it felt like soooooooo long to type), but happy to be convinced to change it back (since my other half of my self thinks sortened names are lazy). but we'll need to follow suit in the product editor code, which at the moment is using `priceSpec`.

ahdinosaur commented on ordering/stories/ProductList.js:
```diff
@@ -0,0 +1,68 @@
+import ProductList from '../components/ProductList'
+
+// https://github.com/root-systems/cobuy/wiki/Models
+const mockProductList = [
```
yay sweet examples! 😊

ahdinosaur commented on ordering/components/ProductPriceSpec.js:
```diff
@@ -0,0 +1,43 @@
+      h(FormattedMessage, {
+        id: 'ordering.priceSpec',
+        className: styles.priceSpecText,
+        values: {
```
this is rad, but shouldn't we pass in `priceSpec.currency` and use that as well in the internationalization message (instead of `$`, which only applies to currencies that use name dollar)? if we want to use a symbol, keen to have a hard-coded object that maps currencies to symbols, like https://github.com/root-systems/cobuy/blob/master/quantity/data/units.js, then use that.

sarah-arrrgh replied:
have changed it so it will show as `NZD7.99` instead of `$7.99`

ahdinosaur commented on ordering/components/SingleViewProduct.js:
```diff
@@ -0,0 +1,74 @@
+  )
+}
+
+const ConnectedSingleViewProduct = reduxForm({ form: 'singleViewProduct' })(SingleViewProduct)
```
hey, the purpose of the `compose` / `pipe` chain for higher-order components is to avoid this. would be much better after `connectFela`. 😄

sarah-arrrgh replied:
done!

ahdinosaur commented on ordering/stories/ProductPriceSpec.js:
```diff
@@ -0,0 +1,35 @@
+
+const ConnectedPriceSpecForm = reduxForm({ form: 'priceSpec' })(PriceSpecForm)
+
+storiesOf('ordering.ProductPriceSpec', module)
```
good work on the story (which includes this boilerplate)!

ahdinosaur commented on ordering/components/ProductFacet.js:
```diff
@@ -0,0 +1,47 @@
+        }, map((facetValue) => {
+          return h(RadioButton, {
+            label: facetValue.name,
+            value: facetValue.name
```
value would be `facetValue.id` right?

sarah-arrrgh replied:
done!

ahdinosaur reviewed (commented):
very legit, nice work! ✨ 

added some feedback 💚

ahdinosaur commented:
@sarah-arrrgh @iainkirkpatrick can we haz a screenshot or gif of what these components look like? keen to capture progress as we go, plus it communicates what this change does without having to spin up the app.

sarah-arrrgh committed:
feat(ordering): style product list view to be slightly less horrid

sarah-arrrgh committed:
feat(ordering): style single product view so it looks slightly less horrid

sarah-arrrgh committed:
feat(ordering): update list view product to show currency instead of himBHsign

sarah-arrrgh committed:
feat(ordering): clean up after mikey's comments on PR

sarah-arrrgh commented:
@ahdinosaur ready for your perusal 🐱

ahdinosaur committed:
fix after dogstack-agents update

https://github.com/root-systems/cobuy/pull/180

ahdinosaur committed:
fixup before merge

- use priceSpecs instead of priceSpecifications: https://github.com/root-systems/cobuy/pull/177#discussion_r138530284
- use {price} {currency} i18n messages
- update dogstack-agents
